provisioner/ansible/provisioner.go

```diff
@@ -118,6 +120,11 @@ func (p *Provisioner) Prepare(raws ...interface{}) error {
 		p.config.AnsibleEnvVars = append(p.config.AnsibleEnvVars, "ANSIBLE_SCP_IF_SSH=True")
 	}
 
+	if p.config.SetPackerPasswd {
+		var PackerEnvVar string = fmt.Sprintf("PACKER_RANDOM_PASSWORD=%s", getWinRMPassword(p.config.PackerBuildName))
```

Comment:
Okay, your problem is that you're trying to get the password here in the Prepare() function; this function is called at the very beginning of the packer run, and the winrm password hasn't been generated yet. You need to set this inside of the Provision() function, which is called after the password has been generated.

Reply:
ahhhh I see. I will finish this over the weekend.